strings/strings.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * Return a number comparing two strings lexicographically.
+ *
+ * @example
+ * ```ts
+ * import { compare } from "https://deno.land/std@$STD_VERSION/strings/strings.ts";
+ *
+ * compare("a", "b"); // -1
+ * compare("a", "a"); // 0
+ * compare("b", "a"); // 1
+ * ```
+ */
+
+export function compare(a: string, b: string): number {
+  if (a === b) {
+    return 0;
+  } else if (a < b) {
+    return -1;
+  } else {
+    return 1;
+  }
+}
+
+/**
+ * Report whether a substring is within a string.
+ *
+ * @example
+ * ```ts
+ * import { contains } from "https://deno.land/std@$STD_VERSION/strings/strings.ts";
+ *
+ * contains("Hello, Deno!", "Deno"); // true
+ * ```
+ */
+
+export function contains(s: string, substr: string): boolean {
+  return s.includes(substr);
+}
+
+/**
+ * Count the number of non-overlapping instances of substr in str. Return 1 + the number
+ * of Unicode points in str if substr is empty.
+ *
+ * @example
+ * ```ts
+ * import { count } from "https://deno.land/std@$STD_VERSION/strings/strings.ts";
+ *
+ * count("cheese", "e"); // 3
+ * count("five", ""); // 5
+ * ```
+ */
+
+export function count(str: string, substr: string): number {
+  if (substr === "") return str.length + 1;
+  return str.split(substr).length - 1;
+}
```

Comment:
This one feels like a homework assignment more than a function someone would actually use. 

Also, I'd probably use a regex

```ts
"cheese".match(/e/g)?.length ?? 0 // 3
```

Not a big fan of having functions that won't really be used.

Reply:
I think it's okay. Seems useful enough to be included in [Voca](https://vocajs.com/#countSubstrings), [Ruby's String class](https://ruby-doc.org/3.2.0/String.html#method-i-count), and most importantly [Golang std](https://pkg.go.dev/strings@go1.19.5#Count).

We should build for performance rather than whatever is most idiomatic. In this case, [`match` is definitely faster than `split`](https://measurethat.net/Benchmarks/Show/7822/0/strmatch-vs-strsplit), but probably hard to beat a naive loop. Could probably look at [Voca](https://github.com/panzerdp/voca/blob/master/src/count/count_substrings.js) or maybe there's some clues in [v8](https://chromium.googlesource.com/v8/v8/+/4.3.49/src/string.js?autodive=0%2F%2F#1018) that we could build on.